auth/services/oauth/oauth.go

```diff
@@ -239,6 +318,54 @@ func (s *service) validateSubject(context *validationContext) error {
 // validate the legal base, according to RFC003 §5.2.1.7 if sid is present
 // use consent store
 func (s *service) validateAuthorizationCredentials(context validationContext) error {
+	// filter on authorization credentials
+	vcs, err := context.verifiableCredentials()
+	if err != nil {
+		return fmt.Errorf(errInvalidVCClaim, err)
+	}
+	j := 0
+	for _, vc := range vcs {
+		if vc.IsType(*credential.NutsAuthorizationCredentialTypeURI) {
+			vcs[j] = vc
+			j++
+		}
+	}
+
+	vcs = vcs[:j]
+
+	// no auth creds, return
+	if len(vcs) == 0 {
+		return nil
+	}
+
+	iat := context.jwtBearerToken.IssuedAt()
+	iss := context.jwtBearerToken.Issuer()
+	sub := context.jwtBearerToken.Subject()
+
+	for _, authCred := range vcs {
+		// first check if the VC is valid
+		if err := s.vcValidator.Validate(authCred, true, &iat); err != nil {
+			return fmt.Errorf(errInvalidVCClaim, err)
+		}
+
+		//The credential issuer equals the sub field of the JWT.
+		if authCred.Issuer.String() != sub {
+			return fmt.Errorf("issuer of authorization credential with ID: %s does not match jwt.sub: %s", authCred.Issuer.String(), sub)
+		}
+
+		//The credential credentialSubject.id equals the iss field of the JWT.
+		authCredSubjects := make([]credential.NutsAuthorizationCredentialSubject, 0)
+		if err := authCred.UnmarshalCredentialSubject(&authCredSubjects); err != nil {
+			return fmt.Errorf(errInvalidVCClaim, err)
+		}
+		// should be only 1 credentialSubject, but we do the range just to make sure and to avoid [0] specific code.
+		for _, authCredSubject := range authCredSubjects {
```

Comment:
can there be 0 subjects (in some odd case/attacker/bug), making this check never run?

Reply:
could be the case, then the actor will be denied access through the access policy at the resource server level.

Reply:
because the access policy is derived from the Verifiable Credential, not from the token?